source/LumiCalReco/src/GlobalMethodsClass.cpp

```diff
@@ -323,3 +313,90 @@ void GlobalMethodsClass::PrintAllParameters() const {
   streamlog_out(MESSAGE) << "---------------------------------------------------------------" << std::endl;
 
 }
+
+void GlobalMethodsClass::SetGeometryGear() {
+
+  // GEAR access
+  // BP. access gear file
+  gear::GearMgr* gearMgr =  marlin::Global::GEAR;
+  const gear::CalorimeterParameters& _lcalGearPars = gearMgr->getLcalParameters();
+  // geometry parameters of LumiCal
+  // inner and outer radii [mm]
+  GlobalParamD[RMin] = _lcalGearPars.getExtent()[0];
+  GlobalParamD[RMax] = _lcalGearPars.getExtent()[1];
+
+  // starting/end position [mm]
+  GlobalParamD[ZStart] = _lcalGearPars.getExtent()[2];
+  GlobalParamD[ZEnd]   = _lcalGearPars.getExtent()[3];
+
+  // cell division numbers
+  GlobalParamD[RCellLength]   = _lcalGearPars.getLayerLayout().getCellSize0(0);
+  GlobalParamD[PhiCellLength] = _lcalGearPars.getLayerLayout().getCellSize1(0);
+  GlobalParamI[NumCellsR]   = (int)( (GlobalParamD[RMax]-GlobalParamD[RMin]) / GlobalParamD[RCellLength]);
+  GlobalParamI[NumCellsPhi] = (int)(2.0 * M_PI / GlobalParamD[PhiCellLength] + 0.5);
+  GlobalParamI[NumCellsZ] = _lcalGearPars.getLayerLayout().getNLayers();
+
+ // beam crossing angle ( convert to rad )
+  GlobalParamD[BeamCrossingAngle] = _lcalGearPars.getDoubleVal("beam_crossing_angle") / 1000.; 
+
+  // layer thickness
+  GlobalParamD[ZLayerThickness] = _lcalGearPars.getLayerLayout().getThickness(0);
+}
+
+
+bool GlobalMethodsClass::SetGeometryDD4HEP() {
+#ifdef FCAL_WITH_DD4HEP
+  std::cout << __PRETTY_FUNCTION__  << std::endl;
+  DD4hep::Geometry::LCDD& lcdd = DD4hep::Geometry::LCDD::getInstance();
+
+  DD4hep::Geometry::DetElement lumical(lcdd.detector("LumiCal"));
+  DD4hep::Geometry::Segmentation readout(lcdd.readout("LumiCalCollection").segmentation());
+
+  streamlog_out(MESSAGE) << "Segmentation Type" << readout.type()  << std::endl;
+  streamlog_out(MESSAGE) <<"FieldDef: " << readout.segmentation()->fieldDescription()  << std::endl;
+
+  const DD4hep::DDRec::LayeredCalorimeterData * theExtension = lumical.extension<DD4hep::DDRec::LayeredCalorimeterData>();
+  const std::vector<DD4hep::DDRec::LayeredCalorimeterStruct::Layer>& layers= theExtension->layers;
+
+  GlobalParamD[RMin] = theExtension->extent[0]/dd4hep::mm;
+  GlobalParamD[RMax] = theExtension->extent[1]/dd4hep::mm;
+
+  // starting/end position [mm]
+  GlobalParamD[ZStart] = theExtension->extent[2]/dd4hep::mm;
+  GlobalParamD[ZEnd]   = theExtension->extent[3]/dd4hep::mm;
+
+  // cell division numbers
+  typedef DD4hep::DDSegmentation::TypedSegmentationParameter< double > ParDou;
```

Comment:
I take the values from the segmentation, partially because I didn't look closely at the DDRec structure. But mostly because I prefer taking the values from the one place where there are surely used the same way as there are needed here. I don't have to take the detour via the DDRec objects to get these values.